lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = """
+---
+module: elasticache_parameter_group
+short_description: Manage elasticache parameter group
+description:
+  - Manage elasticache parameter group
+  - Has the capabilities to create/destroy parameter group
+version_added: "2.2"
+author: "Aditya Kurniawan (@akurniawan)"
+options:
+  pg_family:
+    description:
+      - Family type for parameter groupo
+    required: false
+    choices: ['memcached1.4', 'redis2.6', 'redis2.8', 'redis3.2']
+  pg_desc:
+    description:
+      - Description for parameter group
+    required: false
+    default: False
+  pg_name:
+    description:
+      - Name for parameter groupo
+    required: false
+  pg_params:
+    description:
+      - List of parameters for elasticacheh parameter group (see example)
+    required: True
+  state:
+    description:
+      - Use C(present) to create and modify the cluster
+      - or node inside the cluster
+      - Use C(absent) to destroy the cluster or node inside the cluster
+      - Use C(reboot) to reboot the cluster or node inside the cluster
+    required: true
+"""
+
+EXAMPLES = """
+- elasticache_parameter_group:
+    state: "present"
+    pg_name: "testing-pg"
+    pg_desc: "testing parameter group"
+    pg_family: "memcached1.4"
+    pg_params:
+      lru_maintainer: 1
+
+- elasticache_parameter_group:
+    state: "absent"
+    pg_name: "testing-pg"
+"""
+
+try:
+    import boto3
+    import botocore
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+"""
+Helper functions
+"""
+
+
+def get_es_client(params):
+    """Get elasticache client"""
+    session = boto3.Session(aws_access_key_id=params["aws_access_key"],
```

Comment:
we should be using boto3_conn for all connections. For example, security_token is missing here. boto3_conn avoids you having to worry about that.